File: riak/jython/update/riak_update.py
#! /usr/bin/jython
# -*- coding: utf-8 -*-
#
#	update/riak_update.py
#
#					Aug/21/2013
#
# ----------------------------------------------------------------
import	os
import sys
reload(sys)
sys.setdefaultencoding('utf-8')
import string
import json
import	datetime
#
sys.path.append ('/var/www/data_base/common/jython_common')
from jython_get_uri import get_uri_proc
from jython_get_uri import rest_put_proc
#
# ----------------------------------------------------------------
print ("*** 開始 ***")
#
key = sys.argv[1]
population_in = string.atoi (sys.argv[2])
print ("%s\t%d" % (key, population_in))
#
host = "localhost:8098"
path = "/riak/shimane" + "/" + key
json_str = get_uri_proc (host,path)
#
print json_str
#
#
city = json.loads (json_str)
print city['name']
#
city['population'] = population_in
city['date_mod'] = '%s' % datetime.date.today ()
#
out_json = json.dumps (city)
#
print out_json
#
url_target = "http://" + host + path
rest_put_proc (url_target,out_json,"text/json")
#
print ("*** 終了 ***")
# ----------------------------------------------------------------

File: riak/ringo/update/riak_update.js
#! /usr/bin/ringo
// ------------------------------------------------------------------
//
//	riak_update.js
//
//						Mar/15/2013
//
// ------------------------------------------------------------------
var text_manipulate = require ('/var/www/data_base/common/ringo_common/text_manipulate.js');
// ------------------------------------------------------------------
print	("*** 開始 ***");
var args = require('system').args;
var key_in = args[1];
var population_in = args[2];
print	(key_in + "\t" + population_in);

var client= require ("ringo/httpclient");

var url_target='http://localhost:8098/riak/shimane/' + key_in;

var data=client.get (url_target);

print (data.status);

var city = JSON.parse (data.content);

print (city.name);

city.population = population_in;
city.date_mod = text_manipulate.get_current_date_proc ();
var out_str = JSON.stringify (city);

client.put (url_target,out_str);

print	("*** 終了 ***");
// ------------------------------------------------------------------
